lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1729,6 +1734,167 @@ void NodeOp::getCanonicalizationPatterns(RewritePatternSet &results,
   results.insert<FoldNodeName>(context);
 }
 
+SmallVector<Value> getCompleteWrite(Operation *lhs) {
+  auto lhsTy = lhs->getResult(0).getType();
+  if (!lhsTy.isa<BundleType>() && !lhsTy.isa<FVectorType>())
+    return {};
+
+  DenseMap<uint32_t, Value> fields;
+  for (Operation *user : lhs->getResult(0).getUsers()) {
+    if (user->getParentOp() != lhs->getParentOp())
+      return {};
```

Comment:
Is this possible in FIRRTL Dialect? Or is this specifically avoiding the case involving assignments in `when`s?

Reply:
yes, it's to not touch things prior to when elimination.